accord-core/src/main/java/accord/local/Command.java

```diff
@@ -911,57 +917,50 @@ enum EnsureAction { Ignore, Check, Add, TrySet, Set }
      * Validate we have sufficient information for the route, partialTxn and partialDeps fields, and if so update them;
      * otherwise return false (or throw an exception if an illegal state is encountered)
      */
-    private boolean validate(Ranges existingRanges, Ranges additionalRanges, ProgressShard shard,
-                             Route<?> route, EnsureAction ensureRoute,
+    private boolean validate(Ranges existingRanges, Ranges additionalRanges,
+                             ProgressShard shard, Route<?> route, EnsureAction ensureRoute,
+                             @Nullable PartialTxn partialTxn, EnsureAction ensurePartialTxn,
+                             @Nullable PartialDeps partialDeps, EnsureAction ensurePartialDeps)
+    {
+        return validate(existingRanges, additionalRanges, additionalRanges, shard, route, ensureRoute, partialTxn, ensurePartialTxn, partialDeps, ensurePartialDeps);
+    }
+    private boolean validate(Ranges existingRanges, Ranges additionalRouteRanges, Ranges additionalTxnRanges,
```

Comment:
You'll note the distinction only exists for `accept`, basically, where we prefer to also update our `Route` to include the ranges we would be involved in but we have no additional transaction data to replicate anyway. 

This isn't strictly necessary, as I have changed how we register ourselves with the `CommandStore` so that this `Route` isn't necessary for anything right now. Though it might help avoid future bugs in C* when forwarding data to other nodes, with this `CommandStore` having dependency data for keys not in the `Route`. That said, since we track dependency data by `Seekable/Key` maybe this isn't a concern either, and we should roll this change back and restore the prior documentation, and just not that in the case of `Accept` we might be registered against keys not implicitly covered by the `Route`.

Reply:
Let me know if you have an opinion here. Might be easier to just rollback, but don't have a strong opinion.

Reply:
I don't think I understand how registration changed beyond removing `uncommitted`? It looks basically the same except for the addition of `rangeCommands` to support ranges.

I also need to better understand what you mean by ranges we don't replicate, you mean ranges from older epochs?

Which route isn't necessary, the one passed to accept that is then passed to `Command.set`?

Reply:
I could have been clearer (and perhaps I forgot this at the time), but I believe this is a vestigial change, as I had briefly erroneously used the `Route` to register during `Accept` in an earlier draft of the patch (which is non-sensical, as the types aren't correct). I left it because having a more complete `Route` might all-things-considered be better _but_ I don't foresee us actually persisting the `Route` separately per-store once Aleksey's work lands, permitting us to simply reference a commit log entry and the new command state resulting from processing it, so perhaps better to just roll this back.

Reply:
I have removed this change

Reply:
Ok, so this change might still be desirable, or something like it. I have added logic to fail if we try to set an empty Route, as we _shouldn't_ ordinarily permit it. However, if we only participate in the `executeAt.epoch` for  `accept` (not in the `txnId.epoch`), then without this change we try to set an empty `Route` as we _don't_ want to set any txn or deps contents. We only register for the accept ranges to ensure we are adopted as a dependency.

Perhaps I could just not use the `validate` and `set` logic for `accept`. I'll think about it.